deployment/src/main/java/io/quarkiverse/cxf/deployment/CxfWebServiceBuildItem.java

```diff
@@ -74,4 +75,14 @@ public String getImplementor() {
     public boolean IsClient() {
         return isClient;
     }
+
+    public CXFClientData clientData() {
```

Comment:
The idea behind CXFClientData is to avoid method calls with a large numer of arguments, especially if those arguments do all have the same type.

In this way, CXFClientData is not intended to be JavaBean but rather a Java Record type (introduced in Java 16).

Perhaps a nicer would be to use the builder pattern I guess.

Now what to do. Shall I change into something like

`cxf.setSoapBinding(this.getSoapBinding());`

i.e. simply introduce setters and getters on `CXFClientData()`?

Reply:
Let me add ctor then ..

Reply:
Ok, I changed CXFClientData into a plain and serializable (!) POJO.

Reply:
Hey apologies I could have been more clear on this previously - what I meant is why do we need the clientData method at all in CxfWebServiceBuildItem class?  You should use a CXFClientData constructor wherever this clientData() method is being called, and remove this method all together if possible.

Reply:
I changed it. Nonetheless, don't see any relevant problem. CxfWebServiceBuildItem simply has all the knowledge of creating such an CXFClientData. Now I moved this method into the processor as static function. Fine with me, but I thing that such boiler-plate code is better located in responsible classes while the processor should focus on business logic.